### PR TITLE
✨ Vagrant connection for v9

### PR DESCRIPTION
**Pull request by czunker**

### CODE REVIEW

**Comment by czunker** [comment]
The basic connection to a vagrant vm is working, but something seems off with the platform:
```
 ~/workspace/mondoo/examples  ~/workspace/mondoo/github.com/cnquery/cnquery run vagrant default -c "packages.list.length" 
! using builtin provider for os
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
could not detect suitable package manager for platform
packages.list.length: no data available
 ~/workspace/mondoo/examples  ~/workspace/mondoo/github.com/cnquery/cnquery run vagrant default -c "asset{*}" 
! using builtin provider for os
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
3 errors occured:
	* cannot find 'eol' in resource 'asset'
	* could not detect suitable package manager for platform
	* cannot convert primitive with NO type information
asset: {
  kind: ""
  build: ""
  version: "22.04"
  title: "Ubuntu 22.04.1 LTS"
  arch: "x86_64"
  labels: {}
  platform: "ubuntu"
  vulnerabilityReport: could not detect suitable package manager for platform
  eol: cannot find 'eol' in resource 'asset'
  runtime: ""
  ids: [
    0: "ubuntu-jammy"
    1: "bd38fd7da670484b98768467f85c3004"
  ]
  name: cannot convert primitive with NO type information
  fqdn: ""
  family: [
    0: "debian"
    1: "linux"
    2: "unix"
    3: "os"
  ]
}
```

File: providers/os/config/config.go [hunk 1]
```diff
@@ -115,5 +115,31 @@ var Config = plugin.Provider{
...
+		{
+			Name:    "vagrant",
+			Use:     "vagrant host",
+			Short:   "a Vagrant host",
+			MinArgs: 1,
+			MaxArgs: 1,
+			Flags: []plugin.Flag{
+				/*
+					// looks like we currently do not have sudo support for vagrant
+					// Should we support it?
```
Comment (czunker) [inline providers/os/config/config.go:130]: The current cnquery version does not support it:
```
cnquery run vagrant default --sudo -c 'machine.baseboard{*}'                                                                                                      
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
Query encountered errors:
failed to create resource 'machine.baseboard': could not retrieve smbios info for platform: permission denied
machine.baseboard: no data available
```

**Comment by czunker** [comment]
This works now also with vagrant VMs:
```
cnquery run vagrant default -c "packages.length" 
! using builtin provider for os
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
packages.length: 466
```

File: providers/os/config/config.go (hunk 1 above)
Comment (chris-rock) [reply providers/os/config/config.go:130]: Is the same flag as for local not working?

```
{
	Long:        "sudo",
	Type:        plugin.FlagType_Bool,
	Default:     "false",
	Desc:        "Elevate privileges with sudo.",
	ConfigEntry: "sudo.active",
},
```

Comment (czunker) [reply providers/os/config/config.go:130]: No, something is wrong there. But that's independent of v9. I created an issue to track it: https://github.com/mondoohq/cnquery/issues/1530

**Review by vjeffrey — APPROVED** [review]

**Pull request merged**